wandb/sdk/data_types/base_types/media.py

```diff
@@ -27,6 +27,54 @@
 def _wb_filename(
     key: Union[str, int], step: Union[str, int], id: Union[str, int], extension: str
 ) -> str:
+    r"""Generates a safe filename/path for storing media files, using the provided key, step, and id.
+
+    The filename is made safe by:
+    1. Removing any leading slashes to prevent writing to absolute paths
+    2. Replacing '.' and '..' with underscores to prevent directory traversal attacks
+
+    If the key contains slashes (e.g. 'images/cats/fluffy.jpg'), subdirectories will be created:
+        media/
+          images/
+            cats/
+              fluffy.jpg_step_id.ext
+
+    Args:
+        key: Name/path for the media file
+        step: Training step number
+        id: Unique identifier
+        extension: File extension (e.g. '.jpg', '.mp3')
+
+    Returns:
+        A sanitized filename string in the format: key_step_id.extension
+
+    Raises:
+        ValueError: If running on Windows and the key contains invalid filename characters
+                   (\\, :, *, ?, ", <, >, |)
+    """
+    if SYS_PLATFORM == "Windows" and not util.check_windows_valid_filename(key):
```

Comment:
hmmm... weird question, given that this function is only used here... do you think we should still keep it as a utility or just put it explicit here  (it stems from huge dislike of our long and awful utils. file)

Reply:
Good suggestion, +1 to not having huge util files. So I moved this into the media.py file where it is explicitly used.